opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java

```diff
@@ -105,29 +111,14 @@ public void init(FilterConfig filterConfig) throws ServletException {
         this.filterConfig = filterConfig;
 
         // init if the tracer is null
-        if (tracer == null) {
-            ServletContext servletContext = filterConfig.getServletContext();
-
-            Object contextAttribute = servletContext.getAttribute(TRACER);
-            if (contextAttribute == null) {
-                log.severe("Tracer was not found in `ServletContext.getAttribute(TRACER)`, skipping tracing filter");
-                this.skipFilter = true;
-                return;
-            }
-            if (!(contextAttribute instanceof Tracer)) {
-                log.severe("Tracer from `ServletContext.getAttribute(TRACER)`, is not an instance of " +
-                        "io.opentracing.Tracer, skipping tracing filter");
-                this.skipFilter = true;
-                return;
-            }
-            this.tracer = (Tracer)contextAttribute;
-
-            this.spanDecorators = (List<SpanDecorator>)servletContext.getAttribute(SPAN_DECORATORS);
-            if (this.spanDecorators == null) {
-                this.spanDecorators = Arrays.asList(SpanDecorator.STANDARD_TAGS);
+        if (spanDecorators == null) {
+            spanDecorators = decoratorSPI();
+            if (spanDecorators.isEmpty()) {
+                this.spanDecorators = Arrays.<ServletFilterSpanDecorator>asList(new ServletFilterSpanDecorator.StandardTags());
             }
-            this.spanDecorators.removeAll(Collections.singleton(null));
         }
+
+        log.info("Using servlet filter span decorators:" + spanDecorators);
```

Comment:
Alright, that's forgivable. What is not forgivable is that the source level is still at 1.7, which was EOL'ed almost 2 years ago (!!!). 

And yes, I'll bring that up every time we can't use a lambda :)